validator_client/src/block_service.rs

```diff
@@ -202,16 +235,48 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
             )
         }
 
+        let private_tx_proposals = self.private_tx_proposals;
+        let merge_slot = self
+            .context
+            .eth2_config
+            .spec
+            .bellatrix_fork_epoch
+            .unwrap_or_else(Epoch::max_value)
+            .start_slot(E::slots_per_epoch());
+        let now = self.slot_clock.now().unwrap();
         for validator_pubkey in proposers {
             let service = self.clone();
             let log = log.clone();
             self.inner.context.executor.spawn(
                 async move {
-                    if let Err(e) = service.publish_block(slot, validator_pubkey).await {
+                    //TODO(sean): how should pre-transition and the transition be handled?
+                    let publish_result = if private_tx_proposals && now > merge_slot {
```

Comment:
this seems like a good way to handle the transition, although I would have thought the check should be `now >= merge_slot`?

what problems do you foresee?

Reply:
I was generally worried about outsourcing payload construction at transition, which paul addressed [here](https://github.com/sigp/lighthouse/pull/2987#issuecomment-1028487109) (I haven't implented his suggestion) 

But with updates to serve pre-bellatrix blocks on this endpoint, I think It might be better to just remove any slot logic here.  We could add logic for "don't outsource payloads until post-merge finalization" to the beacon node's `validator/blinded_blocks` endpoint

Reply:
Ah yeah, that sounds sensible. I'm happy to come back to that after merging an MVP version of this PR.